### PR TITLE
chap11

### PR DESCRIPTION
**Pull request by hueanmy**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 
 class User < ApplicationRecord
-  attr_accessor :remember_token
-  before_save :downcase_email
+  attr_accessor :remember_token, :activation_token
+  before_save   :downcase_email
```
Comment (dat-hedspi) [inline app/models/user.rb]: indent, chia block

File: app/models/user.rb [hunk 2]
```diff
@@ -49,6 +51,11 @@ def new_token
 
   private
   def downcase_email
-    self.email = self.email.downcase
+    self.email = email.downcase
+  end
+
+  def create_activation_digest
+    self.activation_token  = User.new_token
```
Comment (dat-hedspi) [inline app/models/user.rb]: indent

File: app/models/user.rb [hunk 3]
```diff
@@ -49,6 +51,11 @@ def new_token
...
+    self.activation_digest = User.digest(activation_token)
```
Comment (dat-hedspi) [inline app/models/user.rb]: bỏ ()

File: app/models/user.rb [hunk 4]
```diff
@@ -17,14 +16,18 @@ class User < ApplicationRecord
 
   has_secure_password
 
+  before_save   :downcase_email
```
Comment (dat-hedspi) [inline app/models/user.rb]: indent

**Comment by dat-hedspi** [comment]
OK

File: app/models/user.rb [hunk 5]
```diff
@@ -17,14 +15,18 @@ class User < ApplicationRecord
...
+
   def remember
     self.remember_token = User.new_token
     update_attributes remember_digest: User.digest(remember_token)
   end
 
   def authenticated? remember_token
-    return false if remember_digest.nil?
-    BCrypt::Password.new(remember_digest).is_password? remember_token
+    digest = send("#{attribute}_digest")
```
Comment (sangnuce) [inline app/models/user.rb]: bỏ ngoặc

File: app/models/user.rb [hunk 6]
```diff
@@ -17,14 +15,18 @@ class User < ApplicationRecord
...
+    return false if digest.nil?
```
Comment (sangnuce) [inline app/models/user.rb]: dùng unless

**Comment by sangnuce** [comment]
OK

**Pull request merged**